File: src/main/java/com/manson/fo76/helper/JFlatCustom.java
package com.manson.fo76.helper;

import com.github.opendevl.JFlat;
import java.io.Writer;

public class JFlatCustom extends JFlat {

  public JFlatCustom(String jsonString) {
    super(jsonString);
  }

  @Override
  public JFlatCustom json2Sheet() {
    return (JFlatCustom) super.json2Sheet();
  }

  @Override
  public JFlatCustom headerSeparator() throws Exception {
    return (JFlatCustom) super.headerSeparator();
  }

  @Override
  public JFlatCustom headerSeparator(String separator) throws Exception {
    return (JFlatCustom) super.headerSeparator(separator);
  }

  public void write2csv(Writer writer, Character delimiter) {
    try {
      boolean comma;
      for (Object[] o : this.getJsonAsSheet()) {
        comma = false;
        for (Object t : o) {
          if (t == null) {
            writer.write(comma ? delimiter.toString() : "");
          } else {
            writer.write(comma ? delimiter + t.toString() : t.toString());
          }
          if (!comma) {
            comma = true;
          }
        }
        writer.write(System.lineSeparator());
      }
      writer.close();
    } catch (Exception e) {
      e.printStackTrace();
    }
  }
}

File: src/main/java/com/manson/fo76/helper/JFlatCustom.kt
package com.manson.fo76.helper

import com.github.opendevl.JFlat
import java.io.Writer

class JFlatCustom(jsonString: String?) : JFlat(jsonString) {
    override fun json2Sheet(): JFlatCustom {
        return super.json2Sheet() as JFlatCustom
    }

    @Throws(Exception::class)
    override fun headerSeparator(): JFlatCustom {
        return super.headerSeparator() as JFlatCustom
    }

    @Throws(Exception::class)
    override fun headerSeparator(separator: String): JFlatCustom {
        return super.headerSeparator(separator) as JFlatCustom
    }

    fun write2csv(writer: Writer, delimiter: Char) {
        try {
            var comma: Boolean
            for (o in jsonAsSheet) {
                comma = false
                for (t in o) {
                    if (t == null) {
                        var str = delimiter.toString()
                        if (!comma) {
                            str = ""
                        }
                        writer.write(str)
                    } else {
                        writer.write(if (comma) delimiter.toString() + t.toString() else t.toString())
                    }
                    if (!comma) {
                        comma = true
                    }
                }
                writer.write(System.lineSeparator())
            }
            writer.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}